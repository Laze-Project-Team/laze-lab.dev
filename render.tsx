import type {
  Queries,
  queries,
  RenderOptions,
  RenderResult,
} from '@testing-library/react';
import { render as defaultRender } from '@testing-library/react';
import type { FC, ReactNode } from 'react';

import { ColorModeProvider } from '@/components/contexts/ColorModeContext';

export type renderType<
  Q extends Queries = typeof queries,
  Container extends Element | DocumentFragment = HTMLElement,
  BaseElement extends Element | DocumentFragment = Container,
> = (
  ui: React.ReactElement,
  options?: RenderOptions<Q, Container, BaseElement>,
) => RenderResult<Q, Container, BaseElement>;

type wrapperProps = {
  children?: ReactNode;
};

type globalWrapperHOCType = (
  wrapper?: React.JSXElementConstructor<{ children: React.ReactElement }>,
) => FC<wrapperProps>;

const globalWrapperHOC: globalWrapperHOCType = (Wrapper) => {
  const globalWrapper: FC<wrapperProps> = ({ children }) => (
    <ColorModeProvider>
      {Wrapper ? (
        <Wrapper>
          <>{children}</>
        </Wrapper>
      ) : (
        children
      )}
    </ColorModeProvider>
  );

  return globalWrapper;
};

export const render: renderType = (ui, options) =>
  defaultRender(ui, {
    ...options,
    wrapper: globalWrapperHOC(options?.wrapper),
  });
