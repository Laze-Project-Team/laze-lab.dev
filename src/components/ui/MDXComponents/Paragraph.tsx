import { css } from '@emotion/react';
import Typography from '@mui/material/Typography';
import type { FC } from 'react';

export const Paragraph: FC<JSX.IntrinsicElements['p']> = ({ children }) => (
  <Typography
    variant="body1"
    css={css`
      margin-bottom: 0.5rem;
      font-size: 0.9rem;
      line-height: 1.5rem;
    `}
  >
    {children}
  </Typography>
);
