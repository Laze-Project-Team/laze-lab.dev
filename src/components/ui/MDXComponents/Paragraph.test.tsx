import '@testing-library/jest-dom';

import { expect, it } from 'vitest';

import { render } from '@/lib/test/render';

import { Paragraph } from './Paragraph';

it('should be rendered correctly', () => {
  const { container } = render(<Paragraph />);
  expect(container.firstChild).toMatchSnapshot();
});
