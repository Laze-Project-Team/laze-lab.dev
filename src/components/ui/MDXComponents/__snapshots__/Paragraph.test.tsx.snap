// Vitest Snapshot v1

exports[`should be rendered correctly 1`] = `
<p
  class="MuiTypography-root MuiTypography-body1 css-1sute3l-MuiTypography-root-Paragraph"
/>
`;
